### ISSUE TITLE
Set customer's preferred locales based on the locale of checkout session

### ISSUE THREAD

**Issue by kouariga** [stripe/stripe-firebase-extensions#491]
# Feature request

- Extension name: `firestore-stripe-payments`

## Is your feature request related to a problem? Please describe.

I would like to set `preferred_locales` for newly created customers based on the browser setting. Currently it is not set, so customers get an invoice in English even though they bought a product from the localized checkout page.

## Describe the solution you'd like

The client-side code can set the locale of the checkout session when adding a `checkout_session` document. Also, the `createCheckoutSession` extension has access to the checkout page's `locale` when creating a customer record. Therefore, we can let the client explicitly set the locale of the checkout session and copy it to the customer's `preferred_locales` on the backend. 

## Describe alternatives you've considered

We could also do the following.
1. Set the Firebase user's custom claim based on the auth's language code, and make the extension read that to set the customer's `preferred_locales`. Personally, I don't think it's a good idea to pollute the user's custom claim for this purpose only.
2. Call Stripe's API directly to set the customer's preferred locale on the client side, based on the browser's setting. This is plausible, but seems like extra work for the user.

**Comment by jsteele-stripe** [comment]
Any Customer objects created by a Checkout Session _should_ have the [`preferred_locales` field](https://stripe.com/docs/api/customers/object#customer_object-preferred_locales) set according the the locale of the session. This is default API behaviour, separate from this extension.

Curious to see a specific example where this was not the case.

**Comment by kouariga** [comment]
The point is that the extension [_creates_ a new Customer object](https://github.com/stripe/stripe-firebase-extensions/blob/477ceef1272c7d2a48afecdc82bc5146bf7043dc/firestore-stripe-payments/functions/src/index.ts#L71) [if it does not exist already](https://github.com/stripe/stripe-firebase-extensions/blob/477ceef1272c7d2a48afecdc82bc5146bf7043dc/firestore-stripe-payments/functions/src/index.ts#L152-L161) (when it creates a Checkout Session for a Firebase user for the first time, for example). In this case, shouldn't it set the customer's `preferred_locales` field to the locale of the session just like Checkout Session does?

This can be done by passing the session's [`locale`](https://github.com/stripe/stripe-firebase-extensions/blob/477ceef1272c7d2a48afecdc82bc5146bf7043dc/firestore-stripe-payments/functions/src/index.ts#L139) parameter to the createCustomerRecord function [here](https://github.com/stripe/stripe-firebase-extensions/blob/477ceef1272c7d2a48afecdc82bc5146bf7043dc/firestore-stripe-payments/functions/src/index.ts#L156).

**Comment by jsteele-stripe** [comment]
@kouariga Yep, I guess in instances where a `customerRecord` does not already exist, then it makes sense to use the `locale` parameter (if passed) to set `preferred_locales` on the created Customer object.

**Comment by kouariga** [comment]
@jsteele-stripe Awesome! Would you like me to create a PR?

**Comment by jsteele-stripe** [comment]
@kouariga Absolutely, if you can!

**Comment by RORO-ML** [comment]
Are there any updates here? Can't find the PR, is implementation still planned?
